source/matrix_free/dof_info.cc

```diff
@@ -994,16 +994,19 @@ namespace internal
           const std::function<void(
             const std::function<void(
               const unsigned int, const unsigned int, const bool)> &)> &loop) {
-          bool all_nodal_and_tensorial = true;
-          for (unsigned int c = 0; c < n_base_elements; ++c)
-            {
-              const auto &si =
-                shape_info(global_base_element_offset + c, 0).data.front();
-              if (!si.nodal_at_cell_boundaries ||
-                  (si.element_type ==
-                   MatrixFreeFunctions::ElementType::tensor_none))
-                all_nodal_and_tensorial = false;
-            }
+          bool all_nodal_and_tensorial = shape_info.size(1) == 1;
+
+          if (all_nodal_and_tensorial)
```

Comment:
While this is not wrong, I wonder if you really need this: We only change the value if we do not have this feature, so if we enter with it as `false`, no harm will be done, except for some additional work. But since this a very cheap operation, I do not see much value in trying to be clever - in that case, we would also need to break the loops below. (I am just trying to keep the number of changed lines down, but I guess you do not want this to be in 9.4 anyway, so then this would also be ok.)

Reply:
The logic below is not really correct. `shape_info(global_base_element_offset + c, 0)` indicates that no hp is used.

Reply:
Would it matter? I mean, we only try to set a variable to `false` that already is false, so I would not have seen anything happening (and a smart compiler seeing that there is no side effect might even remove the unnecessary code, albeit I don't think there are compilers with those abilities today.)